src/util/preprocessData.js

```diff
@@ -345,18 +347,34 @@ function processTimelineEvent(
   }
 }
 
+function preprocessFlamechart(rawData: TimelineEvent[]): FlamechartData {
+  const parsedData = importFromChromeTimeline(rawData, 'react-devtools');
+  const profile = parsedData.profiles[0]; // TODO: Choose the main CPU thread only
+  const flamechart = new Flamechart({
+    getTotalWeight: profile.getTotalWeight.bind(profile),
+    forEachCall: profile.forEachCall.bind(profile),
+    formatValue: profile.formatValue.bind(profile),
+    getColorBucketForFrame: () => 0,
+  });
+  return flamechart;
+}
+
 export default function preprocessData(
   timeline: TimelineEvent[],
 ): ReactProfilerData {
-  const profilerData = {
+  const flamechart = preprocessFlamechart(timeline);
+
```

Comment:
Nit: Might wanna remove this extra line for no reason.

Reply:
This empty line was intentional as I want to separate the profiler data declaration that belongs to the rest of this function from the flamechart computation. Imo they're doing different things conceptually so they should be in separate "paragraphs"